x/evm/evmtest/test_deps.go

```diff
@@ -48,7 +48,13 @@ func (deps TestDeps) NewStateDB() *statedb.StateDB {
 }
 
 func (deps TestDeps) NewEVM() (*vm.EVM, *statedb.StateDB) {
-	stateDB := deps.EvmKeeper.NewStateDB(deps.Ctx, statedb.NewEmptyTxConfig(gethcommon.BytesToHash(deps.Ctx.HeaderHash())))
+	if deps.EvmKeeper.Bank.StateDB == nil {
+		deps.EvmKeeper.Bank.StateDB = deps.NewStateDB()
+	}
+	stateDB := deps.EvmKeeper.Bank.StateDB
+	defer func() {
+		deps.EvmKeeper.Bank.StateDB = nil
+	}()
```

Comment:
You're not necessarily supposed to reset on calls of `NewEVM`, only when you're starting a new `EthereumTx`. In the upgrade handler, it's essentially a transaction, so we need to reset at the end, but not throughout the rest of the codebase uniformly

Reply:
Yep, was debugging this locally and realized that a lot of tests use NewEVM() multiple times, so clearing the StateDB has some issues. I will mark this PR as draft for now and figure out a more elegant solution.